com.unity.ml-agents/Runtime/Agent.cs

```diff
@@ -788,11 +792,15 @@ void AgentStep()
 
         void DecideAction()
         {
-            m_Action.vectorActions = m_Brain?.DecideAction();
-            if (m_Action.vectorActions == null)
+            var action = m_Brain?.DecideAction();
+
+            if (action == null)
```

Comment:
I think ResetData() will actually be a no-op here (since it only updates the buffers if `m_Action.vectorActions` is null). Should we just zero out `m_Action.vectorActions` instead ?